src/Manager.js

```diff
@@ -537,8 +552,30 @@ class GiveawaysManager extends EventEmitter {
     async _init() {
         const rawGiveaways = await this.getAllGiveaways();
         rawGiveaways.forEach((giveaway) => this.giveaways.push(new Giveaway(this, giveaway)));
+        const cacheAllGiveawayChannels = async () => {
+            if (!this.client.startTime) return setTimeout(cacheAllGiveawayChannels, 100);
+            for (const giveaway of this.giveaways) {
+                let channel = this.client.getChannel(giveaway.channelId);
+                if (channel) continue;
+                channel =
+                    (await this.client.getRESTChannel(giveaway.channelId).catch(() => {})) ||
+                    (await this.client.getMessage(giveaway.channelId, giveaway.messageId).catch(() => {}))?.channel;
+                if (!channel?.id) continue;
+                if (!channel.name) {
+                    try {
+                        channel = require('../../eris/lib/structures/Channel').from(await this.client.requestHandler.request('GET', `/channels/${channel.id}`, true), this.client);
+                    } catch {
+                        channel = null;
+                    }
+                }
```

Comment:
To make this cleaner/better.
Temporarily make `client.options.restMode` true, use the rest method, and then return the option to its original state.
Would that be a good "workaround"?